lymph/unilateral.py

```diff
@@ -1052,23 +1016,31 @@ def assign_parameters(self, **kwargs):
             :class:`Marginalizor`) all raise a ``ValueError`` when provided with
             invalid parameters.
         """
-        for key, value in kwargs.items():
-            if key in self.diag_time_dists:
-                try:
-                    self.diag_time_dists[key].update(value)
-                except ValueError as val_err:
-                    raise ValueError(
-                        "Parameters for marginalization over diagnose times are invalid"
-                        ) from val_err
-
-            elif key == 'growth':
-                for edge in self.growth_edges:
-                    edge.spread_prob = value
-            elif key == 'micro_mod':
-                for edge in self.lnl_edges:
-                    edge.micro_mod = value
-            else:
-                self._setter_lookup[key](value)
+        if new_params_list is not None:
+            # here it depends how we want to provide the new_params_list.
+            # there are a lot of options. e.g. one needs to specifically define each base, each transmission, each growth and each micro_mod parameter
+            # Or one only defines each base, each transmission and one growth and one micro_mod parameter
+            # I am not sure what exactly you prefer Roman.
+            # But it kind of needs to be well defined, also how the function handles incomplete parameter lists...

```

Comment:
Agreed, this needs to be well defined.

I think previously, the order was
1. base probs (tumor to LNLs)
2. transmission probs (LNL to LNL)
3. time prior(s)

It would be really beautiful if we could just iterate through the `_setter_lookup` and the provided parameter list like so:
```python
for new_param, setter_func in zip(new_params_list, self._setter_lookup.values()):
    setter_func(new_param)
```
And conversely, a user could print out the order in which parameters are set by just looking at the keys of the setter lookup.

But this would require that we initialize the `_setter_lookup` differently, because e.g. there is no setter function for setting all growth parameters...

Maybe we can leave that unimplemented as of now, I can try something unnecessarily fancy later on 😅

Anyways, you can change the method to look like this:
```python
def assign_params(self, *new_params_args, **new_params_kwargs) -> None:
    """Set params."""
    if len(new_params_args) > 0:
        warnings.warn("Setting params via arguments not yet implemented.")
    # Do all the rest here
```
That way we can later think about it.

Reply:
Sounds good, :)